### ISSUE TITLE
Error

### ISSUE THREAD

**Issue by Bencey** [42wim/matterbridge#567]
I get this error whenever I try to run mattermost -> discord. Any idea what it means
```
PS D:\bots\go-work\bin> ./matterbridge -conf matterbridge.toml
[2018-11-07T15:57:26+13:00]  INFO main:         Running version 1.11.4-dev
[2018-11-07T15:57:26+13:00]  INFO main:         WARNING: THIS IS A DEVELOPMENT VERSION. Things may break.
[0000] FATAL             blah1 error(s) decoding:

* 'Gateway': source data must be an array or slice, got map
```

**Comment by Bencey** [comment]
https://pastebin.com/KaDfNsze

**Comment by jheiselman** [comment]
@Bencey alright, I'm sorry to do this, but pastebin refuses to let me see that paste. Keeps forcing me to solve captchas and telling me that I'm wrong. Can you use https://gist.github.com please?

**Comment by Bencey** [comment]
Sure

**Comment by Bencey** [comment]
https://gist.github.com/Bencey/1d98255aeff8d962c0c9286df212dde1

**Comment by jheiselman** [comment]
Can you please try running with the '-debug' flag and capture that output and post it? Your config looks okay to me.

**Comment by Bencey** [comment]
kk

**Comment by Bencey** [comment]
open -debug: The system cannot find the file specified

**Comment by jheiselman** [comment]
Sorry, run `./mattermost -debug -conf matterbridge.toml`

**Comment by Bencey** [comment]
PS D:\bots\go-work\bin> ./matterbridge -debug -conf matterbridge.toml
[0000]  INFO main:         Enabling debug
[0000]  INFO main:         Running version 1.11.4-dev
[0000]  INFO main:         WARNING: THIS IS A DEVELOPMENT VERSION. Things may break.
[0000] FATAL             blah1 error(s) decoding:

* 'Gateway': source data must be an array or slice, got map

**Comment by Bencey** [comment]
Oh whoops

**Comment by Bencey** [comment]
mattermost or matterbridge?

**Comment by jheiselman** [comment]
No, you were correct. I mistyped. That's not much more information is it? May need to wait for @42wim to chime in. I'm not sure where that error is even coming from. It looks like it would be in the config parser, but it is an error that must be getting passed up from further in.

**Comment by jheiselman** [comment]
Ah. I think I've found it. It appears to be an [issue](https://github.com/spf13/viper/issues/324) in the viper library. I see that you're using a dev version. Can you try with one of the [pre-compiled release versions](https://github.com/42wim/matterbridge/releases/tag/v1.11.3)? Those will have passed testing and are probably using an earlier version of viper that may lack this bug.

**Comment by Bencey** [comment]
Alright I'll try when I get home, thanks
